app/src/main/java/com/vultisig/wallet/ui/screens/vault_settings/VaultSettingsViewModel.kt

```diff
@@ -40,9 +58,30 @@ internal open class VaultSettingsViewModel @Inject constructor(
     fun onEvent(event: VaultSettingsEvent) {
         when (event) {
             Backup -> backupVault()
-            Delete -> deleteVault()
+            ShowConfirmDeleteScreen -> showConfirmDeleteDialog()
+            DismissConfirmDeleteScreen -> dismissConfirmDeleteDialog()
+            Delete -> delete()
             ErrorDownloadFile -> errorBackUp()
             is SuccessBackup -> successBackup(event.fileName)
+            is ChangeCheckCaution -> changeCheckCaution(event.index, event.isChecked)
```

Comment:
Consider exposing and calling this method on ViewModel instead of using events here. They don't seem like they do much of anything.

Reply:
Done